config/config.go

```diff
@@ -665,6 +777,21 @@ var defaultConf = Config{
 		RecordPlanInSlowLog: logutil.DefaultRecordPlanInSlowLog,
 		EnableSlowLog:       *NewAtomicBool(logutil.DefaultTiDBEnableSlowLog),
 	},
+	Instance: Instance{
+		TiDBGeneralLog:              false,
+		EnablePProfSQLCPU:           false,
+		DDLSlowOprThreshold:         DefDDLSlowOprThreshold,
+		ExpensiveQueryTimeThreshold: DefExpensiveQueryTimeThreshold,
+		EnableSlowLog:               *NewAtomicBool(logutil.DefaultTiDBEnableSlowLog),
+		SlowThreshold:               logutil.DefaultSlowThreshold,
+		RecordPlanInSlowLog:         logutil.DefaultRecordPlanInSlowLog,
+		CheckMb4ValueInUTF8:         *NewAtomicBool(true),
+		ForcePriority:               "NO_PRIORITY",
+		MemoryUsageAlarmRatio:       DefMemoryUsageAlarmRatio,
+		EnableCollectExecutionInfo:  true,
+		PluginDir:                   "/data/deploy/plugin",
```

Comment:
Some of the variables cannot be changed by the system variable(at least for now), 'plugin_dir' is one of them. For those variables, I think it's better to make them 'read-only'. Could you update the code to do that and add the related test cases?

Reply:
This makes sense to me too. There is the `ReadOnly: true` attribute you can set on the system variable to achieve this.